src/js/containers/MenuIcon.js

```diff
@@ -24,6 +25,12 @@ const mapStateToProps = (state) => {
 const mapDispatchToProps = (dispatch) => {
     return {
         onSelection: (appID, path) => {
+            const state = store.getState()
+            const app = state.ui[state.activeApp]
+            if (app) {
+                app.activeSubMenu = null
```

Comment:
There is an easier way to do this. Use the dispatch parameter.

```
dispatch(deactivateSubMenu(appID))
```

Reply:
That should replace all changes here in this onSelection

Reply:
Be sure to import the action

```
import { deactivateSubMenu } from '../actions'
```

Reply:
good call! changed in https://github.com/smartdevicelink/generic_hmi/pull/179/commits/a51b2fb95d7f7620d01a9566135817583345a4d1